cache/cache.go

```diff
@@ -44,3 +45,19 @@ type LocalArtifactCache interface {
 	// Clear deletes the local database
 	Clear() (err error)
 }
+
+// isStale checks if the cache is stale or not.
+// e.g. When {"alpine": 1, "python": 2} is cached and {"alpine": 2, "python": 2} is sent, the cache is stale.
+// Also, {"python": 2} cache must be replaced by {"alpine": 1, "python": 2}.
+func isStale(current, cached map[string]int) bool {
+	for analyzerType, currentVersion := range current {
+		cachedVersion, ok := cached[analyzerType]
```

Comment:
In this case, can't determine stale.
The isStale `return false` .
```
current: map[ amazon:1 ]
cached: map[ amazon:1, alpine:1 ]
```

Reply:
It is intended as it is explained in the description.

Reply:
Thanks, make sense.